src/smo/data/note.ts

```diff
@@ -120,6 +125,10 @@ export interface SmoNoteParams {
    * note duration
    */
   ticks: Ticks,
+  /**
+   * visible duration
+   */
+  stemTicks: number,
```

Comment:
I _think_ I like what you've done here.  The main issue will be keeping this up-to-date with actual ticks, as durations change.  The original idea was to have measure, which knows about triplets, decide if the ticks here represented the stem length or not.